components/tools/OmeroJava/test/integration/LightAdminRolesTest.java

```diff
@@ -2437,4 +2453,182 @@ public void testModifyGroupEdit(boolean isPrivileged,
+        ContrastMethod ho = new ContrastMethodI();
+        ho.setValue(omero.rtypes.rstring("testDeleteEnumerationsbyRestrictedSystemUser"));
+        final ITypesPrx ts = root.getSession().getTypesService();
+        List<IObject> types = ts.allEnumerations(ContrastMethod.class.getName());
+        int n = types.size();
+        ho = (ContrastMethod) ts.createEnumeration(ho);
+        types = ts.allEnumerations(ContrastMethod.class.getName());
+        Assert.assertEquals(types.size(), (n+1));
+        logNewAdminWithoutPrivileges();
+        //try to delete an enum type. The type is not important
+        final ITypesPrx types_svc = factory.getTypesService();
+        types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        int m = types.size();
+        Assert.assertEquals(m, (n+1));
+        Iterator<IObject> i = types.iterator();
+        while (i.hasNext()) {
+            IObject o = i.next();
+            if (o.getId().getValue() == ho.getId().getValue()) {
+                types_svc.deleteEnumeration(ho);
+                break;
+            }
+        }
+        types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        Assert.assertEquals(types.size(), n);
+    }
+
+    /**
+     * Tests if an enumeration already used can be deleted.
+     * An exception should be thrown.
+     * @throws Exception
+     */
+    @Test(expectedExceptions = omero.ValidationException.class)
+    public void testDeleteUsedEnumerationsbyRestrictedSystemUser() throws Exception {
+        logNewAdminWithoutPrivileges();
+        //try to delete an enum type. The type is not important
+        final ITypesPrx types_svc = factory.getTypesService();
+        List<IObject> types = types_svc.allEnumerations(EventType.class.getName());
+        Iterator<IObject> i = types.iterator();
+        while (i.hasNext()) {
+            types_svc.deleteEnumeration(i.next());
+        }
+    }
+
+    /**
+     * Resets the enumerations
+     * @throws Exception
+     */
+    @Test
+    public void testResetEnumerationsbyRestrictedSystemUser() throws Exception {
+        logNewAdminWithoutPrivileges();
+        ContrastMethod ho = new ContrastMethodI();
+        ho.setValue(omero.rtypes.rstring("testResetEnumerationsbyRestrictedSystemUser"));
+        final ITypesPrx types_svc = factory.getTypesService();
+        List<IObject> types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        //original number of enumerations
+        int n = types.size();
+        ho = (ContrastMethod) types_svc.createEnumeration(ho);
+        types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        //check that an enumeration has been added
+        Assert.assertEquals(types.size(), (n+1));
+        types_svc.deleteEnumeration(ho);
+        types_svc.resetEnumerations(ContrastMethod.class.getName());
+        types = types_svc.allEnumerations(ContrastMethod.class.getName());
+        Assert.assertEquals(types.size(), n);
```

Comment:
And the size after delete + restore is again "n" ? I would expect "n+1" if the reset really restores the deleted enum, as indicated in the comment https://github.com/openmicroscopy/openmicroscopy/pull/5502/files#diff-d91232ef1e245a87e311c4a79a2a5f77R2459 in testDeleteEnum....

Reply:
it resets to the original as far as I know

Reply:
No, you are right, the test passes for me locally as it is. Probably still needs some redesign as discussed.